src/features/wallet/helpers/checkAvailableRewards.ts

```diff
@@ -0,0 +1,27 @@
+import { ethers } from "ethers";
+import { VoteRevealed } from "web3/get/queryVotesRevealedEvents";
+import { queryRetrieveRewards } from "web3/get/queryRetrieveRewards";
+
+export default async function checkAvailableRewards(
+  data: VoteRevealed[],
+  address: string,
+  contract: ethers.Contract
+) {
+  const promises = data.map(async (vote) => {
+    const rewardAvailable = await queryRetrieveRewards(
+      contract,
+      address,
+      vote.roundId,
+      vote.identifier,
+      vote.time
+    );
+    if (rewardAvailable) return parseFloat(rewardAvailable);
+  });
+
+  const values = await Promise.all(promises);
+
+  return values.reduce((balance, val) => {
+    if (val) return (balance ? balance : 0) + val;
+    return balance;
+  }, 0);
```

Comment:
Can this be simplified or am I missing something? I assume you should be able to tell typescript that balance _must_ be defined, right?
```suggestion
  return values.reduce((balance, val) => balance + val ? val : 0, 0);
```